src/youtube-player/youtube-player.ts

```diff
@@ -0,0 +1,216 @@
+  private _destroyed = new EventEmitter<undefined>();
+
+  private _player: Player | undefined;
+
+  constructor() {
+    const playerObs =
+      createPlayerObservable(
+        this._youtubeContainer,
+        this._videoId,
+      ).pipe(waitUntilReady(), takeUntil(this._destroyed), publish());
+
+    /** Set up side effects to bind inputs to the player. */
+    playerObs.subscribe(player => this._player = player);
+
+    bindCueVideoCall(playerObs, this._videoId, this._destroyed);
+
+    // After all of the subscriptions are set up, connect the observable.
+    (playerObs as ConnectableObservable<Player>).connect();
+  }
+
+  ngAfterViewInit() {
+    if (!this.youtubeContainer) {
+      return;
+    }
+    this._youtubeContainer.emit(this.youtubeContainer.nativeElement);
+  }
+
+  ngOnDestroy() {
+    if (!this._player) {
+      return;
+    }
+    this._player.destroy();
+    this._destroyed.emit();
+  }
+}
+
+function waitUntilReady()
+    : OperatorFunction<UninitializedPlayer | undefined, Player | undefined> {
+  return flatMap(player => {
+      if (!player) {
+        return observableOf<Player|undefined>(undefined);
+      }
+      if ('getPlayerStatus' in player) {
+        return observableOf(player as Player);
+      }
+      // The player is not initialized fully until the ready is called.
+      return fromPlayerOnReady(player)
+          .pipe(first(), startWith(undefined));
+    });
+}
+
+/** Since removeEventListener is not on Player when it's initialized, we can't use fromEvent. */
+function fromPlayerOnReady(player: UninitializedPlayer): Observable<Player> {
+  return new Observable<Player>(emitter => {
+    let aborted = false;
+
+    const onReady = (event: YT.PlayerEvent) => {
+      if (aborted) {
+        return;
+      }
+      event.target.removeEventListener('onReady', onReady);
+      emitter.next(event.target);
+    };
+
+    player.addEventListener('onReady', onReady);
+
+    return () => {
+      aborted = true;
+    };
+  });
+}
+
+/** Create an observable for the player based on the given options. */
+function createPlayerObservable(
+  youtubeContainer: Observable<HTMLElement>,
+  videoIdObs: Observable<string | undefined>,
+): Observable<UninitializedPlayer | undefined> {
+
+  const playerOptions =
+    videoIdObs
+    .pipe(
+      map((videoId) => videoId ? {videoId} : undefined),
+    );
+
+  return combineLatest(youtubeContainer, playerOptions)
+      .pipe(scan(syncPlayerState, undefined), distinctUntilChanged());
+}
+
+/** Destroy the player if there are no options, or create the player if there are options. */
+function syncPlayerState(
+  player: UninitializedPlayer | undefined,
+  [container, videoOptions]: [HTMLElement, YT.PlayerOptions | undefined],
+  _: number,
+): UninitializedPlayer | undefined {
+  if (!videoOptions) {
+    if (player) {
+      player.destroy();
+    }
+    return;
+  }
+  if (!player) {
+    const newPlayer: UninitializedPlayer = new YT.Player(container, videoOptions);
```

Comment:
Out of curiosity: do we need to somehow check that the `YT` global variable is available? otherwise I could expect some weird confusing errors if that's not the case.